src/Microsoft.VisualStudio.ProjectSystem.Managed.VS.Client/ProjectProperties/ProjectPropertiesEditorFactory.cs

```diff
@@ -0,0 +1,168 @@
+
+            // An empty caption allows the project name to be used as the caption.
+            pbstrEditorCaption = "";
+
+            pguidCmdUI = default;
+            pgrfCDW = default;
+
+            if ((grfCreateDoc & (VSConstants.CEF_OPENFILE | VSConstants.CEF_SILENT)) == 0)
+            {
+                Debug.Fail("Must be opening the file (not cloning it) and creating the editor silently");
+                return HResult.InvalidArg;
+            }
+
+            _paneData = new ProjectPropertiesWindowPaneData(pvHier, ServiceProvider.GlobalProvider);
+
+            ppunkDocData = Marshal.GetIUnknownForObject(_paneData);
+
+            var newEditor = new ProjectPropertiesWindowPane(pvHier);
+            ppunkDocView = Marshal.GetIUnknownForObject(newEditor);
+
+            const _VSRDTFLAGS windowFlags = _VSRDTFLAGS.RDT_DontAddToMRU |    // TODO review this
+                                            _VSRDTFLAGS.RDT_CantSave |        // TODO review this -- we do want to save
+                                            _VSRDTFLAGS.RDT_ProjSlnDocument |
+                                            _VSRDTFLAGS.RDT_VirtualDocument |
+                                            _VSRDTFLAGS.RDT_DontAutoOpen;     // TODO review this
+
+            pgrfCDW = (int)windowFlags;
+
+            pbstrEditorCaption = string.Empty; // no need to add anything to the caption
+                
+            return HResult.OK;
+        }
+
+        public int MapLogicalView(ref Guid rguidLogicalView, out string? pbstrPhysicalView)
+        {
+            // The default physical view for an editor must be null, and we only support one physical view.
+            pbstrPhysicalView = null;
+
+            // Disallow TextView, as that suggests the requested view is the XML editor.
+            if (rguidLogicalView == VSConstants.LOGVIEWID.TextView_guid)
+            {
+                return HResult.NotImplemented;
+            }
+
+            if (rguidLogicalView == VSConstants.LOGVIEWID.Primary_guid)
+            {
+                return HResult.OK;
+            }
+
+            if (rguidLogicalView == VSConstants.LOGVIEWID.Designer_guid)
+            {
+                return HResult.OK;
+            }
+
+            // We may also be called with other GUIDs for specific pages in the UI. Rather than list them all here,
+            // just return success.
+            // TODO determine whether we can handle those GUIDs in the new experience or not and potentially change this to return NotImplemented.
+            return HResult.OK;
+        }
+
+        public int Close()
+        {
+            return HResult.OK;
+        }
+
+        private sealed class ProjectPropertiesWindowPaneData
+        {
+            public ProjectPropertiesWindowPaneData(IVsHierarchy vsHierarchy, System.IServiceProvider serviceProvider)
+            {
+                
+            }
+        }
+
+        private sealed class ProjectPropertiesWindowPane : WindowPane
```

Comment:
There's async support for window panes, should we be using it? I don't know much about it.

Reply:
As far as I can tell there's async support for tool windows, but not editor/designer windows.